### DISCUSSION TITLE
How can I avoid code duplications in modify templates?

### DISCUSSION THREAD

**Discussion by r-darwish** [twpayne/chezmoi#3519] [Q&A] [answered]
I want to manage [k9s](k9scli.io/) configuration through chezmoi. 

k9s takes its configuration file from different paths depending on the operating system, so I want to do the trick from [here](https://www.chezmoi.io/user-guide/manage-machine-to-machine-differences/) where I put two template files and a common template in `.chezmoitemplates`. However, since k9s modifies its configuration file externally I also need to use a modify script or a modify template in order to manage it. 

I tried going with a modify template, but it seems that modify templates load the templates in `.chezmoitemplates` like regular template. How can I avoid making two copies of the modify template (one in ~/.config and one in ~/Library)?

**Comment by Lockszmith-GH** [comment, marked as answer]
For now, see [#3485](https://github.com/twpayne/chezmoi/discussions/3485#discussioncomment-8146750), tom mentiones that #2273 will eventually resolve this.
